File: frontend/src/api/legal.ts
import { API_BASE_URL } from '../config'

export type LegalMeta = {
  privacy_version: string
  terms_version: string
  consent_version: string
  operator_name: string
  operator_email: string
}

export async function fetchLegalMeta(): Promise<LegalMeta> {
  const res = await fetch(`${API_BASE_URL}/legal/meta`)
  if (!res.ok) throw new Error(`legal meta: ${res.status}`)
  return (await res.json()) as LegalMeta
}

export type RegisterPayload = {
  email: string
  password: string
  accept_privacy: boolean
  accept_terms: boolean
  privacy_version: string
  terms_version: string
}

export async function downloadAccountExport(token: string): Promise<void> {
  const res = await fetch(`${API_BASE_URL}/account/export`, {
    headers: { Authorization: `Bearer ${token}` },
  })
  if (!res.ok) throw new Error(`export: ${res.status}`)
  const blob = await res.blob()
  const url = URL.createObjectURL(blob)
  const a = document.createElement('a')
  a.href = url
  a.download = 'near-data-export.json'
  a.click()
  URL.revokeObjectURL(url)
}

export async function deleteAccount(token: string, password: string): Promise<void> {
  const res = await fetch(`${API_BASE_URL}/account`, {
    method: 'DELETE',
    headers: {
      Authorization: `Bearer ${token}`,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ password, confirmation: 'DELETE' }),
  })
  if (!res.ok) {
    const body = await res.json().catch(() => ({}))
    const detail = typeof body.detail === 'string' ? body.detail : `HTTP ${res.status}`
    throw new Error(detail)
  }
}

File: backend/app/schemas/legal.py
"""Схемы юридических документов и регистрации с согласием."""

from pydantic import BaseModel, Field, field_validator


class LegalMetaRead(BaseModel):
    privacy_version: str
    terms_version: str
    consent_version: str
    operator_name: str
    operator_email: str


class RegisterWithConsent(BaseModel):
    email: str = Field(..., max_length=320)
    password: str = Field(..., min_length=8, max_length=128)
    accept_privacy: bool
    accept_terms: bool
    privacy_version: str = Field(..., min_length=1, max_length=32)
    terms_version: str = Field(..., min_length=1, max_length=32)

    @field_validator("accept_privacy", "accept_terms")
    @classmethod
    def must_accept(cls, v: bool) -> bool:
        if not v:
            raise ValueError("Требуется согласие")
        return v


class DeleteAccountBody(BaseModel):
    password: str = Field(..., min_length=1, max_length=128)
    confirmation: str = Field(..., min_length=1, max_length=32)

    @field_validator("confirmation")
    @classmethod
    def must_confirm(cls, v: str) -> str:
        if v.strip().upper() != "DELETE":
            raise ValueError('Введите DELETE для подтверждения')
        return v
